jobs/update_hosts_last_check_in.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python
+import os
+import sys
+from functools import partial
+from logging import Logger
+
+from connexion import FlaskApp
+from sqlalchemy import text
+from sqlalchemy.orm import Session
+
+from app.environment import RuntimeEnvironment
+from app.logging import get_logger
+from app.models import Host
+from jobs.common import excepthook
+from jobs.common import job_setup
+
+PROMETHEUS_JOB = "inventory-update-hosts-last-check-in"
+LOGGER_NAME = "update-hosts-last-check-in"
+RUNTIME_ENVIRONMENT = RuntimeEnvironment.JOB
+
+LIMIT = int(os.getenv("HOST_UPDATE_LIMIT", 50000))
+
+
+def run(logger: Logger, session: Session, application: FlaskApp):
+    with application.app.app_context():
+        num_hosts_null_last_check_in = session.query(Host).filter(Host.last_check_in is None).count()
+
+        if num_hosts_null_last_check_in > 0:
+            logger.info(f"There are still {num_hosts_null_last_check_in} to be updated")
+            logger.info(f"Updating {LIMIT} hosts' last_check_in field")
+            session.execute(
+                text(
+                    "UPDATE hbi.hosts h SET last_check_in  = sub.modified_on "
+                    f"FROM (SELECT id, modified_on FROM hbi.hosts LIMIT {LIMIT}) AS sub "
+                    "WHERE h.id = sub.id AND h.last_check_in is NULL;"
```

Comment:
Do you think we could simplify this to something along the lines of:
```suggestion
                    "UPDATE hbi.hosts h SET h.last_check_in  = h.modified_on "
                    f"WHERE hbi.hosts.id IN (SELECT id FROM hbi.hosts sub WHERE sub.last_check_in is NULL LIMIT {LIMIT});"
```

I'm not sure, but my gut tells me it might be bit more efficient, we could `EXPLAIN ANALYZE` both through gabi and see, might not be a big deal tho :)

Reply:
I will try to run it locally, cause it is not possible to perform UPDATE query using Gabi.

Let me see what I get here.

Reply:
Plase, refer to this analyses: https://docs.google.com/document/d/1CGf-wmvOcpkE72aQRC_urxeYU7_G40JKwwsB7ybovkA/edit?pli=1&tab=t.0#heading=h.g6t9tutzhu0a

But generally, looks like that the query that we have in place runs faster them the one you suggested!

Appreciate this suggestion, I will add this `EXPLAIN ANALYZE` to my toolbox!

Reply:
Reevaluated the query with @ezr-ondrej, and there is no significant difference in this from each of the queries.

I will would go with your suggestion which is simpler in syntax